src/main/java/doge/Storage.java

```diff
@@ -0,0 +1,119 @@
+        try {
+            FileWriter fw = new FileWriter("./data/doge.txt");
+            for (Task curr : tasks) {
+                fw.write(System.lineSeparator() + curr.toString());
+            }
+            fw.close();
+        } catch (IOException e) {
+            throw new DogeException("Write to storage file failed!");
+        }
+    }
+
+    /**
+     * Loads the tasks from local storage.
+     *
+     * @return the list of tasks
+     * @throws DogeException if it fails to create a new storage file or the parsing of data failed
+     */
+    public ArrayList<Task> load() throws DogeException {
+        try {
+            Path p = Paths.get("./data");
+            Files.createDirectories(p);
+        } catch (IOException e) {
+            throw new DogeException("Failed to create new directory!");
+        }
+
+        try {
+            File f = new File( "./data/doge.txt");
+            ArrayList<Task> temp = new ArrayList<>();
+
+            if (f.createNewFile()) {
+                System.out.println("<NOTICE> Storage file does not exist! Creating one right now...");
+            } else {
+                Scanner s = new Scanner(f);
+                while (s.hasNextLine()) {
+                    String[] curr = s.nextLine().split( "\\|");
+                    String taskStatus = curr.length > 1 ? curr[1].trim() : null;
+                    String[] dateTime;
+                    Task currTask;
+                    Date date;
+                    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+
+                    switch(curr[0].trim()) {
+                    case "T":
+                        currTask = new Todo(curr[2].trim());
+                        if (taskStatus.equals("✓")) {
+                            currTask.mark();
+                        }
+                        temp.add(currTask);
+                        break;
+                    case "D":
+                        dateTime = curr[3].trim().substring(9).trim().split(" ");
+                        date = new SimpleDateFormat("dd-MMM-yyyy").parse(dateTime[0]);
+                        currTask = new Deadline(curr[2].trim(), Doge.getDateTime(sdf.format(date) + " " + dateTime[1]));
+                        if (taskStatus.equals("✓")) {
+                            currTask.mark();
+                        }
+                        temp.add(currTask);
+                        break;
+                    case "E":
+                        dateTime = curr[3].trim().substring(3).trim().split(" ");
+                        date = new SimpleDateFormat("dd-MMM-yyyy").parse(dateTime[0]);
+                        currTask = new Event(curr[2].trim(), Doge.getDateTime(sdf.format(date) + " " + dateTime[1]));
+                        if (taskStatus.equals("✓")) {
+                            currTask.mark();
+                        }
+                        temp.add(currTask);
+                        break;
+                    }
```

Comment:
The `load` method seems quite long to me. Do you think it will be possible to abstract this functionality out to another method, how about the `createLoadedTask `method